src/primary_names.lua

```diff
@@ -321,10 +338,19 @@ function primaryNames.getPaginatedPrimaryNames(cursor, limit, sortBy, sortOrder)
 	local primaryNamesArray = {}
 	local cursorField = "name"
 	for owner, primaryName in pairs(primaryNames.getUnsafePrimaryNameOwners()) do
+		if not primaryName.processId then
+			-- TODO: After initial deployment, we can make processId not nil and remove this block
+			local record = arns.getRecord(baseNameForName(primaryName.name))
+			if record then
+				primaryName.processId = record.processId
+				PrimaryNames.owners[owner] = primaryName
```

Comment:
rather than storing this in state, we can compute it on read using the requested name to get the base name

Reply:
this seemed like a lot of getRecord calls when dealing with a paginated getPrimaryNames call returning 1000 names for example

Reply:
fetching records is cheap (o(1)) since they are indexed by name. we could memoize the process id for a given base name if we wanted to get really efficient

Reply:
thinking further - memoizing wouldn't give us any improvements, and actually increase the memory footprint - disregard